sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/IoTHubConnectionSample.java

```diff
@@ -0,0 +1,504 @@
+                // when we call reactor.stop() later on.
+                Schedulers.boundedElastic().schedule(() -> reactor.run());
+
+                return handler;
+            }),
+            handler -> getConnectionString(handler, properties, entityPath + "/$management"),
+            handler -> {
+                // After we're done fetching a compatible Event Hubs connection string, stop the reactor.
+                reactor.stop();
+                return handler.closeAsync();
+            })
+            // Cache the result of this operation so additional downstream subscribers can make use of the value
+            // instead of us having to create another reactor.
+            .cache();
+
+        // Leverage Mono.usingWhen so the producer client is disposed of after we finish using it.
+        // In production, users would probably cache the Mono's result, reusing the EventHubProducerAsyncClient and
+        // finally closing it.
+        final Mono<EventHubProperties> runOperation = Mono.usingWhen(
+            connectionStringMono.map(connectionString -> {
+                System.out.println("Acquired Event Hubs compatible connection string.");
+
+                return new EventHubClientBuilder()
+                    .connectionString(connectionString)
+                    .buildAsyncProducerClient();
+            }),
+            producer -> {
+                System.out.println("Created producer client.");
+
+                return producer.getEventHubProperties();
+            },
+            producer -> Mono.fromRunnable(() -> {
+                System.out.println("Disposing of producer client.");
+                producer.close();
+            }));
+
+        // Blocking here to turn this into a synchronous operation because we no longer need asynchronous operations.
+        final EventHubProperties eventHubProperties = runOperation.block();
+        if (eventHubProperties == null) {
+            System.err.println("No properties were retrieved.");
+            return;
+        }
+
+        final String partitionIds = eventHubProperties.getPartitionIds()
+            .stream()
+            .collect(Collectors.joining(", "));
+
+        System.out.printf("Event Hub Name: [%s]. Created At: %s. partitionIds: [%s]%n", eventHubProperties.getName(),
+            eventHubProperties.getCreatedAt(), partitionIds);
+    }
+
+    /**
+     * Mono that completes with the corresponding Event Hubs connection string.
+     *
+     * @param handler Proton-J handler.
+     * @param properties Connection string properties.
+     * @param entityPath Entity to connect to.
+     *
+     * @return A Mono that completes when the connection string is retrieved. Or errors if the transport, connection, or
+     *     link could not be opened.
+     */
+    private static Mono<String> getConnectionString(ProtonJHandler handler, IoTConnectionStringProperties properties,
```

Comment:
I suggest we add an public overload that accepts an IoT connection string and returns an EventHub connection string. User's code will use this new function that hides the detailed proton-j code.
The new function may use some code from the "main" method so "main" will simply call this new function.
Both sync and async clients can use the result from this new function.

Reply:
Good call! I'll go and move most of the code.